src/compiler/scanner.ts

```diff
@@ -2469,6 +2468,15 @@ export function createScanner(languageVersion: ScriptTarget, skipTrivia: boolean
                 else if (ch === CharacterCodes.closeBracket) {
                     inCharacterClass = false;
                 }
+                else if (
+                    ch === CharacterCodes.openParen
+                    && charCodeUnchecked(p + 1) === CharacterCodes.question
+                    && charCodeUnchecked(p + 2) === CharacterCodes.lessThan
+                    && charCodeUnchecked(p + 3) !== CharacterCodes.equals
+                    && charCodeUnchecked(p + 3) !== CharacterCodes.exclamation

```

Comment:
I missed this in my last review. These need to be range checked:
```suggestion
                    && charCodeChecked(p + 1) === CharacterCodes.question
                    && charCodeChecked(p + 2) === CharacterCodes.lessThan
                    && charCodeChecked(p + 3) !== CharacterCodes.equals
                    && charCodeChecked(p + 3) !== CharacterCodes.exclamation
```